connect/cli/plugins/product/sync/general.py

```diff
@@ -102,32 +103,24 @@ def sync(self):
         product['category']['id'] = self._category
         # Solution for v22 to avoid issue while updating capabilities
         del product['capabilities']['subscription']['change']['editable_ordering_parameters']
-        try:
-            self._client.products[self._product_id].update(product)
-        except ClientError as e:
-            errors.append(
-                f'Error while updating general product information: {str(e)}',
-            )
-            return errors
-        payload = MultipartEncoder(
-            fields={
-                'icon': (
-                    ws['B9'].value,
-                    open(
-                        os.path.join(
-                            self._media_path,
-                            'media',
-                            ws['B9'].value,
-                        ),
-                        "rb",
-                    ),
-                ),
-            },
+        product['customer_ui_settings'] = json.dumps(product['customer_ui_settings'])
+        data = {
+            'body': (None, json.dumps(product), 'application/json'),
+        }
+        image_name = ws['B9'].value
+        image_data = open(
+            os.path.join(
+                self._media_path,
+                'media',
+                ws['B9'].value,
+            ),
+            "rb",
         )
+        image_type, _ = guess_type(image_name)
+        data['icon'] = (image_name, image_data, image_type)
```

Comment:
Let's make this code a little bit more declarative like you did at [sync/media.py](https://github.com/cloudblue/connect-cli/pull/155/files#diff-83d3c4bfb3357fa228800e0bd2e168bdd0b0bb8d22aa1ab26475f542870d54d6R80)

```suggestion
       # all calculation of necessary parameters before
       ...
        data = {
            'body': (None, json.dumps(product), 'application/json'),
            'icon': (image_name, image_data, image_type),
        }
```

Reply:
It make sense. Done!